eng/pointergen/Generator.cs

```diff
@@ -0,0 +1,898 @@
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static bool operator ==({{name}} lh, NullPtr rh) => lh == ({{name}})rh;
+
+    /// <summary>
+    /// Determines whether a <see cref="{{name}}"/> and a NullPtr are not equal
+    /// </summary>
+    /// <param name="lh"></param>
+    /// <param name="rh"></param>
+    /// <returns>Whether the <see cref="{{name}}"/> and NullPtr are not equal</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static bool operator !=({{name}} lh, NullPtr rh) => lh != ({{name}})rh;
+
+    /// <summary>
+    /// Determines whether a NullPtr and a <see cref="{{name}}"/> are equal
+    /// </summary>
+    /// <param name="lh"></param>
+    /// <param name="rh"></param>
+    /// <returns>Whether the NullPtr and <see cref="{{name}}"/> are equal</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static bool operator ==(NullPtr lh, {{name}} rh) => ({{name}})lh == rh;
+
+    /// <summary>
+    /// Determines whether a NullPtr and a <see cref="{{name}}"/> are not equal
+    /// </summary>
+    /// <param name="lh"></param>
+    /// <param name="rh"></param>
+    /// <returns>Whether the NullPtr and <see cref="{{name}}"/> are not equal</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static bool operator !=(NullPtr lh, {{name}} rh) => ({{name}})lh != rh;
+
+    /// <summary>
+    /// Creates a <see cref="{{name}}"/> from a span
+    /// </summary>
+    /// <param name="span"></param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static implicit operator {{name}}(Span<byte> span) => new(ref span.GetPinnableReference());
+
+    /// <summary>
+    /// Creates a <see cref="{{name}}"/> from a byte pointer
+    /// </summary>
+    /// <param name="ptr"></param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public unsafe static implicit operator {{name}}(byte* ptr) => new(ref Unsafe.AsRef<byte>(ptr));
+
+    /// <summary>
+    /// Creates a <see cref="{{name}}"/> from a void pointer
+    /// </summary>
+    /// <param name="ptr"></param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public unsafe static implicit operator {{name}}(void* ptr) => new(ref Unsafe.AsRef<byte>(ptr));
+
+    /// <summary>
+    /// Creates a byte pointer from a <see cref="{{name}}"/>
+    /// </summary>
+    /// <param name="ptr"></param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public unsafe static implicit operator byte*({{name}} ptr) => (byte*)Unsafe.AsPointer({{getObjectRef}});
```

Comment:
```suggestion
    public unsafe static explicit operator byte*({{name}} ptr) => (byte*)Unsafe.AsPointer({{getObjectRef}});
```
Unsafe.AsPointer is unsafe and should be an explicit op.